yakut/param/formatter.py

```diff
@@ -66,21 +69,99 @@ def _make_json_formatter() -> Formatter:
     return lambda data: typing.cast(str, json.dumps(data, ensure_ascii=False, separators=(",", ":")))
 
 
+def _insert_format_specifier(items: typing.List[typing.Tuple[str, typing.Any]], key, instance, is_start: bool = True):
+    is_list = isinstance(instance, Collection) and not isinstance(instance, str)
+    is_dictionary = isinstance(instance, Mapping)
+    if is_start:
+        if is_dictionary:
+            items.append((key + "{", "{"))
+        elif is_list:
+            items.append((key + "[", "["))
+    else:
+        if is_dictionary:
+            items.append((key + "}", "}"))
+        elif is_list:
+            items.append((key + "]", "]"))
+
+
+def flatten_start(
```

Comment:
1. This is not part of the module API so it should start with an underscore.

2. I don't understand the purpose of invoking an empty closure inside this function. Why not remove the outer function and promote the closure to a top-level function?

Reply:
done